### ISSUE TITLE
[Screen Reader - Bot Framework Web Chat - Markdown]: Screen reader is not announcing the complete article information when the focus lands on the response article.

### ISSUE THREAD

**Issue by pgopinadhu** [microsoft/BotFramework-WebChat#4725]
### **User Experience:**
Screen reader users gets affected as they cannot know the complete article information if the screen reader doesn't announce and stops narrating till the link. Screen reader users miss the rest of the information and may not effectively navigate and access the control.
### **Note:**
User credentials should NOT be included in the bug.
### **Repro Steps:**

1. Launch URL: https://microsoft.github.io/BotFramework-WebChat/01.getting-started/a.full-bundle/ on edge dev.
2. Web chat page opens.
3. Press 4-tab keys to reach 'Type your message' field. Enter the "markdown" keyword on it. Press 1-tab key to reach the 'send' button and activate it with the enter key. Turn on Screen reader (Narrator).
4. Press 2-shift tabs and 1-down arrow to reach the received card/ response
6. Verify whether the screen reader announces the complete information or not.
### **Actual Result:**
Screen reader does not announce the complete article information when the focus lands on the response article.
**Observation:**

- The narrator announces as "Bot said: Basic formatting Paragraphs can be written like so. A paragraph is the basic block of Markdown. A paragraph is what text will turn into when there is no reason it should become anything else. Paragraphs must be separated by a blank line. ...URLs can be made in a handful of ways: A named link to MarkItDown".  -Narrator doesn't announce as 'click to interact also'.
- JAWS announces as "Bot said: Basic formatting Paragraphs can be written like so. A paragraph is the basic block of Markdown. A paragraph is what text will turn into when there is no reason it should become anything else. Paragraphs must be separated by a blank line. ...URLs can be made in a handful of ways: A named link to [MarkItDown][3]. The easiest way to do these is to select what you want to make a link and hit Ctrl ArticleRegion". 
- NVDA announces as "Bot said: Basic formatting Paragraphs can be written like so. A paragraph is the basic block of Markdown. A paragraph is what text will turn into when there is no reason it should become anything else. Paragraphs must be separated by a blank line. ...URLs can be made in a handful of ways: A named link to [MarkItDown][3]. The easiest way to do these is to select what you want to make a link and hit Ctrl+L. Another named link to MarkItDownOpens in a new window; external. Some links have [query strings](https://bing.com?q=someOpens in a new window; external. value) that need encoding Sometimes you just want a URL like http://www.markitdown.net/Opens in a new window; external.. Horizontal rule A horizontal rule is a line that goes across the middle of the page. It’s sometimes handy for breaking things up. Table |header1|header 2| |----|----| | cell 1 | cell 2| | cell three | cell four| Whitespace Here’s a line. This has the standard two newlines before it. This has four newlines before it. This has six newlines before ...This has two newlines and four <br/> tags before it. One or more links in the message. article".

### **Expected Result:**
Screen reader should announce the complete article information when the focus lands on the response article.
### **Note:**

- Issue reproes with Narrator and JAWS.
- Issue doesn't repro with NVDA.
### **MAS Reference:_**
[MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)
### **Reference Links:**

- [Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-poyad%40microsoft.com%7cc11478591b5a474f6d9d08d7199726c0%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637006009300301864&sdata=ZTWYamT1yE7yko5QU35ZJLf/5ZydtDeGAKrQGUiugJs%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier.
- External Bug Process: If this bug belongs to external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [Link](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDV8MDF8fDNkZjhhMWU0NTA0YTQwM2MyNGRhMDhkYWY1NTkwMDNlfDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDB8MHw2MzgwOTIwNTk4MzEzMTM2MjZ8VW5rbm93bnxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazkwYUdWeUlpd2lWMVFpT2pFeGZRPT18MXxNVFkzTXpZd09URTRNall3TXpzeE5qY3pOakE1TVRneU5qQXpPekU1T2pjMllUWXlNRE5pTnpZeFpqUmlObVE0T1dKbE0yRm1aVFZpTjJZd1pHSmhRSFJvY21WaFpDNTJNZz09fDBiZDAxMWU1M2M0YjQ0NGYyNGRhMDhkYWY1NTkwMDNlfGI0NDRlYzhjODQ4YzRiODI4Mzk1NmRjNDdlZjQ3NDky&sdata=dlJVM3lpMWQrRmtpc09HVTI0REh4NlVrU00yMG5xd0JlUFdBOHZuaHNOVT0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-pgopinadhu%40microsoft.com&OR=Teams-HL&CT=1673616420039&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiIyNy8yMzAxMDUwOTQwMCIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D). “HCL Staff should not log any third-party external bugs. Should be routed to edad team”.
- Please reach out to [C&AI Teams channel](https://nam06.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cv-dhti%40microsoft.com%7cf90880a914be401871f108d763233116%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637086876782361405&sdata=/ULS%2BbeqMjysCoiWHvFgAbSnuFJqxO3wtpHmlidPjbU%3D&reserved=0) for any process related queries.
### **Test Environment:**

- OS: Windows 11 Enterprise 22H2 (OS build: 22621.1555)
- Browser: Edge Version 114.0.1807.6 (Official build) dev (64-bit)
- URL: [Web Chat: Full-featured bundle (microsoft.github.io)](https://microsoft.github.io/BotFramework-WebChat/01.getting-started/a.full-bundle/)
- Tools: Developer Tool, Accessibility Insights for Web
- Matrix: Edge dev+ Narrator.

**Comment by pgopinadhu** [comment]
#A11yMAS;#A11ySev3;#MAS1.3.1;#HCL;#Accessibility;#Benchmark;
#Win11-Edge(Chromium);#AILimited;#BotFrameworkWebChat-May23;

**Comment by compulim** [comment]
The following is my investigation notes.

I also reproduce the issue in Windows Narrator that the screen reader stopped narrating after "make a link and hit control".

This DOM element is what screen reader will read when the focus land on the bubble. It contains the full plain text of the Adaptive Card.

<img width="732" alt="image" src="https://github.com/microsoft/BotFramework-WebChat/assets/1622400/3b07828c-eaaa-43e3-83aa-5d64aaa187af">

**Comment by compulim** [comment]
Edge will only send narrate the first 1,000 characters in a text. Thus, it stopped at that position.

I tried the following transcript:

```
Iteration 1. A quick brown fox jumped over the lazy dogs.
Iteration 2. A quick brown fox jumped over the lazy dogs.
Iteration 3. A quick brown fox jumped over the lazy dogs.
Iteration 4. A quick brown fox jumped over the lazy dogs.
Iteration 5. A quick brown fox jumped over the lazy dogs.
Iteration 6. A quick brown fox jumped over the lazy dogs.
Iteration 7. A quick brown fox jumped over the lazy dogs.
Iteration 8. A quick brown fox jumped over the lazy dogs.
Iteration 9. A quick brown fox jumped over the lazy dogs.
Iteration 10. A quick brown fox jumped over the lazy dogs.
Iteration 11. A quick brown fox jumped over the lazy dogs.
Iteration 12. A quick brown fox jumped over the lazy dogs.
Iteration 13. A quick brown fox jumped over the lazy dogs.
Iteration 14. A quick brown fox jumped over the lazy dogs.
Iteration 15. A quick brown fox jumped over the lazy dogs.
Iteration 16. A quick brown fox jumped over the lazy dogs.
Iteration 17. A quick brown fox jumped over the lazy dogs.
Iteration 18. A quick brown fox jumped over the lazy dogs.
Iteration 19. A quick brown fox jumped over the lazy dogs.
Iteration 20. A quick brown fox jumped over the lazy dogs.
```

It stopped right at "Iterat" of "Iteration 18", which is the 1,000-character mark.

**Comment by compulim** [comment]
Overall, I think this is not an issue on Web Chat side. But an issue on Windows Narrator and JAWS with their 1,000-character limitation.
